core/codegen/debug.go

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codegen
+
+import (
+	"debug/dwarf"
+	"fmt"
+	"llvm/bindings/go/llvm"
+	"path/filepath"
+)
+
+// dbg holds debug info generation data.
+type dbg struct {
+	m     *Module
+	cu    llvm.Metadata
+	llvm  *llvm.DIBuilder
+	files map[string]file
+	tys   map[Type]llvm.Metadata
+}
+
+// builder constructs and returns a new llvm debug info builder for the module.
+func (d *dbg) builder() *llvm.DIBuilder {
+	if d.llvm != nil {
+		return d.llvm
+	}
+
+	b := llvm.NewDIBuilder(d.m.llvm)
+	d.cu = b.CreateCompileUnit(llvm.DICompileUnit{
+		Language:       0x0001, // DW_LANG_C,
+		File:           "apis",
+		Dir:            "/ssd/src/gapid", // TODO: HACK
```

Comment:
do we want this here?

Reply:
Debug generation shouldn't be enabled (except you spotted it is below). I haven't figured out a nice way to get the project root here due to bazel complications. I'd like to revisit this once I have performance in the bag.